api/client/network.go

```diff
@@ -222,7 +224,7 @@ func consolidateIpam(subnets, ranges, gateways []string, auxaddrs map[string]str
 				return nil, fmt.Errorf("multiple overlapping subnet configuration is not supported")
 			}
 		}
-		iData[s] = &network.IPAMConfig{Subnet: s, AuxAddress: map[string]string{}}
+		iData[s] = &network.IPAMConfig{Subnet: s, AuxAddress: map[string]string{}, Options: options}
```

Comment:
I agree however this change allows sending subnet specific options for ipam drivers via the API. I personally do not have a use case where per subnet options is useful but I am sure there is one out there. 

However if we look at the current UI when you create a new network via the CLI it only allows you to create one subnet. So since the CLI is already limited in that aspect it doesn't seem like it is a major issue.